test/integration/Subscription.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { getWeb3, getContractAddresses, getOptions, query } from "./util";
+
+const Reputation = require('@daostack/arc/build/contracts/Reputation.json');
+
+const { execute } = require('apollo-link');
+const { WebSocketLink } = require('apollo-link-ws');
+const { SubscriptionClient } = require('subscriptions-transport-ws');
+const ws = require('ws');
+
+const getWsClient = function(wsurl) {
+  const client = new SubscriptionClient(
+    wsurl, {reconnect: true}, ws
+  );
+  return client;
+};
+
+const createSubscriptionObservable = (wsurl, query, variables) => {
+  const link = new WebSocketLink(getWsClient(wsurl));
+  return execute(link, {query: query, variables: variables});
+};
+
+//const gql = require('graphql-tag');
+
+describe('Subscriptions', () => {
+    let web3, addresses, opts, reputation;
+    beforeAll(async () => {
+        web3 = await getWeb3();
+        addresses = getContractAddresses();
+        opts = await getOptions(web3);
+        reputation = new web3.eth.Contract(Reputation.abi, addresses.Reputation, opts);
+    });
+    it('Reputation Mint', async () => {
+        const accounts = web3.eth.accounts.wallet;
+        const SUBSCRIBE_QUERY = `
+        subscription  {
+          reputationMints {
+            contract
+            amount
+            address
+          }
+        }
+        `;
+
+      const subscriptionClient =await  createSubscriptionObservable(
+          'http://graph-node:8001/by-name/daostack', // GraphQL endpoint
+          SUBSCRIBE_QUERY,                                       // Subscription query
+          {address: accounts[0].address.toLowerCase()}                                                // Query variables
+        );
+
+      let event;
+
+      var consumer =await subscriptionClient.subscribe(eventData => {
```

Comment:
I think its easier to do `await subscriptionClient.toPromise()` which just returns the first event as a promise.

Reply:
Not sure how

Reply:
you just call `toPromise` on the observable.